packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -486,31 +486,47 @@ class _AnimatedFadeOutFadeIn extends ImplicitlyAnimatedWidget {
   _AnimatedFadeOutFadeInState createState() => _AnimatedFadeOutFadeInState();
 }
 
-class _AnimatedFadeOutFadeInState extends ImplicitlyAnimatedWidgetState<_AnimatedFadeOutFadeIn> {
-  Tween<double>? _targetOpacity;
-  Tween<double>? _placeholderOpacity;
+class _AnimatedFadeOutFadeInState extends State<_AnimatedFadeOutFadeIn> with SingleTickerProviderStateMixin<_AnimatedFadeOutFadeIn> {
+  late final AnimationController controller = AnimationController(
+    duration: duration,
+    debugLabel: kDebugMode ? widget.toStringShort() : null,
+    vsync: this,
+  );
+  late final Animation<double> animation = CurvedAnimation(parent: controller, curve: Curves.linear);
+  final Tween<double> _targetOpacity = Tween<double>(begin: 0.0, end: 1.0);
+  final Tween<double> _placeholderOpacity = Tween<double>(begin: 1.0, end: 0.0);
   Animation<double>? _targetOpacityAnimation;
   Animation<double>? _placeholderOpacityAnimation;
+  late Duration duration = widget.fadeOutDuration + widget.fadeInDuration;
```

Comment:
why do you need a seperate duration field? Cann you just use controller.duration?